core/network/impl/peer_manager_impl.cpp

```diff
@@ -404,17 +404,18 @@ namespace kagome::network {
 
   void PeerManagerImpl::disconnectFromPeer(const PeerId &peer_id) {
     auto it = active_peers_.find(peer_id);
+    if (peer_id == own_peer_info_.id) {
+      return;
+    }
     if (it != active_peers_.end()) {
       SL_DEBUG(log_, "Disconnect from peer {}", peer_id);
       stream_engine_->del(peer_id);
+      peer_states_.erase(peer_id);
+      host_.disconnect(peer_id);
```

Comment:
In this case peer is disconnected only if it in list of active. Need to check, should we disconnect partially connected peer, or should not.